server/game/gameSystems/DiscardCardSystem.ts

```diff
@@ -0,0 +1,47 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, Location } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardProperties extends ICardTargetSystemProperties {}
+
+export class DiscardCardSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, IDiscardProperties> {
+    public override readonly name = 'discardCard';
+    public override readonly eventName = EventName.OnCardsDiscarded;
+
+    public override canAffect(card: Card | Card[], context: TContext, additionalProperties: Record<string, any> = {}): boolean {
+        /* TODO A bug currently leads to this sometimes being an array. Once the bug is fixed, we can expect this to
+        accept only 1 card, so we should change this to match */
+        const cards = Array.isArray(card) ? card : [card];
+
+        return cards.every((c) => c.location !== Location.Discard && super.canAffect(c, context, additionalProperties));
+    }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        return [
+            'discard {0}',
+            [properties.target]
+        ];
+    }
+
+    public override queueGenerateEventGameSteps(events: GameEvent[], context: TContext, additionalProperties: Record<string, any> = {}): void {
```

Comment:
Typically systems that only target one card don't need their own implementation of `queueGenerateEventGameSteps` because it's handled in the superclass. Was something specific happening that made you add this (and does it work without it?)